### PR TITLE
Handle multi-doc yaml for flux build

### PR DESCRIPTION
**Pull request by SomtochiAma**
Fixes: #2780 

Signed-off-by: Somtochi Onyekwere <somtochionyekwere@gmail.com>

### CODE REVIEW

**Comment by stefanprodan** [comment]
@SomtochiAma there is a typo in the commit message, should be `check` instead of `cheeck`.

File: internal/build/build.go [hunk 1]
```diff
@@ -250,12 +251,19 @@ func (b *Builder) unMarshallKustomization() (*kustomizev1.Kustomization, error)
...
+	for !(k.Name == b.name && (k.Namespace == b.namespace || k.Namespace == "")) {
+		err = decoder.Decode(k)
+		if err != nil {
+			if err == io.EOF {
+				return nil, fmt.Errorf("failed find kustomization with name '%s' and namespace '%s' in file '%s'",
+					b.name, b.namespace, b.kustomizationFile)
+			} else {
+				return nil, fmt.Errorf("failed to unmarshall kustomization file %s: %w", b.kustomizationFile, err)
+			}
+		}
```
Comment (stefanprodan) [inline internal/build/build.go:266]: We should return here to stop the iteration.

Comment (SomtochiAma) [reply internal/build/build.go:266]: It decodes all valid Kubernetes YAML successfully. 
if we find the correct the correct YAML, for loop wouldn't run and the YAML is returned on the next line after the loop

Comment (stefanprodan) [reply internal/build/build.go:266]: Got it! thanks

File: internal/build/testdata/local-kustomization/multi-doc-valid.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+kind: Kustomization
+metadata:
+  name: infra-namespace
+  namespace: flux-system
+  labels:
+    component.kutara.io/part-of: definitions
+spec:
+  path: "./k8s/base/infra"
+  prune: true
+---
```
[the remark this answers is not in the document]
Comment (SomtochiAma) [reply internal/build/testdata/local-kustomization/multi-doc-valid.yaml:12]: Like a second one?

**Review by stefanprodan — APPROVED** [review]
LGTM 

Thanks @SomtochiAma please squash and rebase.

**Pull request merged**